### PR TITLE
Refactor fulfillments quantity fulfilled

### PR DESCRIPTION
**Pull request by tomaszszymanski129**
I want to merge this change because it refactor waiting fulfillments quantity logic.

🚫  Previous logic:
- waiting fulfillments DO NOT count in the line quantity_fulfilled property.
- special treatment for this particular fulfillment type.
- quantity to fulfill required special calculation based on fulfillment lines with waiting ones.

✅  Refactored:
- on orderFulfil (mutation to create fulfillments) they are counted in quantity_fulfilled regardless of state
- treated like any other fulfillment on fulfillment mutations (except cancel, we don't need to deallocate stock)

⚠️ NOTE FOR REVIEWERS: please also check the included data migration.

# Impact

* [ ] New migrations
* [ ] New/Updated API fields or mutations
* [ ] Deprecated API fields or mutations
* [ ] Removed API types, fields, or mutations
* [ ] Documentation needs to be updated

# Pull Request Checklist

* [ ] Privileged queries and mutations are guarded by proper permission checks
* [ ] Database queries are optimized and the number of queries is constant
* [ ] Database migration files are up to date
* [ ] The changes are tested
* [ ] GraphQL schema and type definitions are up to date
* [ ] Changes are mentioned in the changelog

### CODE REVIEW

File: saleor/order/actions.py [hunk 1]
```diff
@@ -350,6 +350,14 @@ def cancel_waiting_fulfillment(
     events.fulfillment_canceled_event(
         order=fulfillment.order, user=user, app=app, fulfillment=None
     )
+
+    order_lines = []
```
Comment (korycins) [inline saleor/order/actions.py:354]: make sure that allocation are correctly returned

File: saleor/order/actions.py [hunk 2]
```diff
@@ -452,19 +460,24 @@ def clean_mark_order_as_paid(order: "Order"):
...
     """Fulfill order line with given quantity."""
-    lines_to_decrease_stock = get_order_lines_with_track_inventory(order_lines_info)
-    if lines_to_decrease_stock:
-        decrease_stock(lines_to_decrease_stock, manager)
-    order_lines = []
-    for line_info in order_lines_info:
-        line = line_info.line
-        line.quantity_fulfilled += line_info.quantity
-        order_lines.append(line)
+    if decrease_stocks:
```
Comment (korycins) [inline saleor/order/actions.py:469]: Why do we need theese `ifs`?

**Review by korycins — COMMENTED** [review]
@tomaszszymanski129 What about migrating existing fulfillments to a new approach? I believe we should move them, right?

**Comment by tomaszszymanski129** [comment]
@korycins good point! I will prepare data migration.

**Review by korycins — APPROVED** [review]

**Review by IKarbowiak — APPROVED** [review]

**Review by maarcingebala — APPROVED** [review]

**Pull request merged**